ingress-lb/kontena.yml

```diff
@@ -0,0 +1,64 @@
+stack: kontena/ingress-lb
+version: 0.1.0
+description: Generic ingress load balancer for http/https
+expose: lb
+variables:
+  lb_affinity:
+    type: string
+    default: label!=no-ingress-lb
+    from:
+      prompt: Affinity
+  lb_certs:
+    type: array
+    required: false
+    from:
+      vault_cert_prompt: Pick SSL cert(s) from Vault
+  lb_stats_password:
+    type: string
+    from:
+      vault: LB_STATS_PASSWORD
+      random_string: 12
+    to:
+      vault: LB_STATS_PASSWORD
+  lb_custom_ports:
+    type: array
+    from:
+      prompt: Custom port mappings (comma separated, eg 2022:2022,25:25)
+services:
+  lb:
+    image: kontena/lb:latest
+    ports:
+      - 80:80
+      # {% if lb_certs %}
+      - 443:443
+      # {% endif %}
+      # {% if lb_custom_ports %}
+      # {% for port in lb_custom_ports %}
+      - {{ port }}
+      # {% endfor %}
+      # {% endif %}
+    # {% if lb_affinity %}
+    affinity:
+      - {{ lb_affinity }}
+    # {% endif %}
+    deploy:
+      strategy: daemon
+      wait_for_port: 80
+      interval: 7d # re-deploy once in a week for security updates
+    health_check:
+      protocol: http
+      port: 80
+      uri: /__health
+    environment:
```

Comment:
Maybe make this optional as there has been some issues with this with e.g. AWS ELBs